inference-engine/thirdparty/clDNN/runtime/ocl/ocl_command_queues_builder.cpp

```diff
@@ -12,79 +12,78 @@ namespace ocl {
+                break;
+            case priority_mode_types::low:
+                cl_queue_priority_value = CL_QUEUE_PRIORITY_LOW_KHR;
+                break;
+            default:
+                break;
+        }
+
+        properties.insert(properties.end(), {CL_QUEUE_PRIORITY_KHR, cl_queue_priority_value});
     }
 
-    unsigned cl_queue_priority_value = CL_QUEUE_PRIORITY_MED_KHR;
-
-    switch (_priority_mode) {
-        case priority_mode_types::high:
-            cl_queue_priority_value = CL_QUEUE_PRIORITY_HIGH_KHR;
-            break;
-        case priority_mode_types::low:
-            cl_queue_priority_value = CL_QUEUE_PRIORITY_LOW_KHR;
-            break;
-        default:
-            break;
+    if (_throttle_mode != throttle_mode_types::disabled) {
+        unsigned cl_queue_throttle_value = CL_QUEUE_THROTTLE_MED_KHR;
+        switch (_throttle_mode) {
+            case throttle_mode_types::high:
+                cl_queue_throttle_value = CL_QUEUE_THROTTLE_HIGH_KHR;
+                break;
+            case throttle_mode_types::low:
+                cl_queue_throttle_value = CL_QUEUE_THROTTLE_LOW_KHR;
+                break;
+            default:
+                break;
+        }
+
+        properties.insert(properties.end(), {CL_QUEUE_THROTTLE_KHR, cl_queue_throttle_value});
     }
 
-    unsigned cl_queue_throttle_value = CL_QUEUE_THROTTLE_MED_KHR;
-
-    switch (_throttle_mode) {
-        case throttle_mode_types::high:
-            cl_queue_throttle_value = CL_QUEUE_THROTTLE_HIGH_KHR;
-            break;
-        case throttle_mode_types::low:
-            cl_queue_throttle_value = CL_QUEUE_THROTTLE_LOW_KHR;
-            break;
-        default:
-            break;
+    if (_supports_queue_families) {
+        cl_uint num_queues = 0;
+        cl_uint family = 0;
+
+        std::vector<cl_queue_family_properties_intel> qfprops = device.getInfo<CL_DEVICE_QUEUE_FAMILY_PROPERTIES_INTEL>();
+        for (cl_uint q = 0; q < qfprops.size(); q++) {
+            if (qfprops[q].capabilities == CL_QUEUE_DEFAULT_CAPABILITIES_INTEL && qfprops[q].count > num_queues) {
+                family = q;
+                num_queues = qfprops[q].count;
+            }
+        }
+
+        if (num_queues)
+            properties.insert(properties.end(), {CL_QUEUE_FAMILY_INTEL, family,
+                                                 CL_QUEUE_INDEX_INTEL, stream_id % num_queues});
     }
 
+    cl_command_queue_properties cl_queue_properties =
+        ((_profiling ? CL_QUEUE_PROFILING_ENABLE : 0) | (_out_of_order ? CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE : 0));
+
+    properties.insert(properties.end(), {CL_QUEUE_PROPERTIES, cl_queue_properties, 0});
+
+    return properties;
+}
+
+ocl_queue_type command_queues_builder::build(const cl::Context& context, const cl::Device& device) {
+    ocl_queue_type queue;
     cl_int error_code = CL_SUCCESS;
+    static uint16_t stream_id = 0;
```

Comment:
Shouldn't it be atomic to avoid data races in case of parallel LoadNetwork calls?

Reply:
Thanks, fixed